src/v/storage/compaction_reducers.cc

```diff
@@ -37,8 +36,12 @@ compaction_key_reducer::operator()(compacted_index::entry&& e) {
     using stop_t = ss::stop_iteration;
     const model::offset o = e.offset + model::offset(e.delta);
 
-    auto it = _indices.find(e.key);
```

Comment:
I wonder if transparent hashing would be beneficial here, from the performance perspective

Reply:
TIL, thanks.

Reply:
right, i had a look at the code and byte_hasher is already transparent, plus the e.key type is already the same type